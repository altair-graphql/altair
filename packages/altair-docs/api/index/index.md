# index

## Index

| Member | Description |
| :------ | :------ |
| [registerPluginClass](functions/registerPluginClass.md) | - |
