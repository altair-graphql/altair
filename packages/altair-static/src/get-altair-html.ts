import { readFileSync } from 'fs';
import { resolve } from 'path';
import { getDistDirectory } from './get-dist';

export default function getAltairHtml() {
  return readFileSync(resolve(getDistDirectory(), 'index.html'), 'utf8');
}
