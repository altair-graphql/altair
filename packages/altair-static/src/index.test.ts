import { readFileSync } from 'fs';
import { resolve } from 'path';
import { renderAltair, renderInitialOptions } from './index';
import * as getAltairHtml from './get-altair-html';

const translateRenderedStrToObj = (result: string) => {
  const resultObj = Function(`
  let __options;
  const AltairGraphQL = { init: (options) => { __options = options; } };
  ${result}
  return __options;`)();

  return resultObj;
};

describe('renderInitialOptions', () => {
  it('should return expected string', () => {
    const result = renderInitialOptions({
      baseURL: '/',
      initialQuery: `query {
        Hello
      }`,
      endpointURL: 'https://example.com/graphql',
      initialHeaders: {
        'X-GraphQL-Token': 'asd7-237s-2bdk-nsdk4',
      },
      initialSettings: {
        theme: 'dark',
      },
    });

    expect(translateRenderedStrToObj(result)).toEqual({
      initialQuery: `query {
        Hello
      }`,
      endpointURL: 'https://example.com/graphql',
      initialHeaders: {
        'X-GraphQL-Token': 'asd7-237s-2bdk-nsdk4',
      },
      initialSettings: {
        theme: 'dark',
      },
    });

    expect(result).toMatchSnapshot();
  });
  it('should render boolean values correctly', () => {
    const result = renderInitialOptions({
      preserveState: false,
    });

    expect(translateRenderedStrToObj(result)).toEqual({
      preserveState: false,
    });
  });
});

describe('renderAltair', () => {
  it('should return expected string', () => {
    (getAltairHtml as any).default = jest.fn();
    (getAltairHtml as any).default.mockReturnValue(
      readFileSync(resolve(__dirname, 'index.html'), 'utf8')
    );
    expect(
      renderAltair({
        baseURL: '/',
        initialQuery: `query {
        Hello
      }`,
        endpointURL: 'https://example.com/graphql',
        initialVariables: '{ variable: 1 }',
      })
    ).toMatchSnapshot();
  });
});
