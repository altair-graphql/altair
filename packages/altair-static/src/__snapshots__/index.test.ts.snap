// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderAltair should return expected string 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>Altair</title>
    <base href="/">
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <link rel="icon" type="image/x-icon" href="favicon.ico" />
    <link href="[% STYLES_FILE %]" rel="stylesheet" />
  </head>

  <body>
    <app-root>
      <style>
        .loading-screen {
          /*Prevents the loading screen from showing until CSS is downloaded*/
          display: none;
        }
      </style>
      <div class="loading-screen styled">
        <div class="loading-screen-inner">
          <div class="loading-screen-logo-container">
            <img src="assets/img/logo_350.svg" alt="Altair" />
          </div>
          <div class="loading-screen-loading-indicator">
            <span class="loading-indicator-dot"></span>
            <span class="loading-indicator-dot"></span>
            <span class="loading-indicator-dot"></span>
          </div>
        </div>
      </div>
    </app-root>
    <script
      rel="preload"
      as="script"
      type="text/javascript"
      src="[% RUNTIME_SCRIPT %]"
    ></script>
    <script
      rel="preload"
      as="script"
      type="text/javascript"
      src="[% POLYFILLS_SCRIPT %]"
    ></script>
    <script
      rel="preload"
      as="script"
      type="text/javascript"
      src="[% MAIN_SCRIPT %]"
    ></script>
  <script>
        AltairGraphQL.init({
initialQuery: \`query {
        Hello
      }\`,
endpointURL: \`https://example.com/graphql\`,
initialVariables: \`{ variable: 1 }\`,
});
    </script></body>
</html>
"
`;

exports[`renderInitialOptions should return expected string 1`] = `
"
        AltairGraphQL.init({
initialQuery: \`query {
        Hello
      }\`,
endpointURL: \`https://example.com/graphql\`,
initialHeaders: {"X-GraphQL-Token":"asd7-237s-2bdk-nsdk4"},
initialSettings: {"theme":"dark"},
});
    "
`;
